### PR TITLE
Corne with i2c Pimoroni trackball installed

### PR DESCRIPTION
**Pull request by art4ride**
This PR includes a default Corne layout with the Pimoroni trackball mounted on the right side support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: drivers/sensors/pimoroni_trackball.c [hunk 1]
```diff
@@ -16,7 +16,9 @@
  */
 #include "pimoroni_trackball.h"
 #include "i2c_master.h"
+#ifdef TRACKBALL_DEBUG
 #include "print.h"
+#endif
```
Comment (zvecr) [inline drivers/sensors/pimoroni_trackball.c:21]: Given the nop stubs <https://github.com/qmk/qmk_firmware/blob/master/quantum/logging/print.h#L52-L59>, there should be no need for this change.

```suggestion
#include "print.h"
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/crkbd/keymaps/art4ride_pimoroni_trackball/config.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#pragma once
+
+// #define USE_MATRIX_I2C
+
+/* Select hand configuration */
+
+#define MASTER_RIGHT
+// #define EE_HANDS
+
+#define USE_SERIAL_PD2
```
Comment (drashna) [inline keyboards/crkbd/keymaps/art4ride_pimoroni_trackball/config.h:30]:
```suggestion
#define USE_SERIAL
```

File: keyboards/crkbd/keymaps/art4ride_pimoroni_trackball/config.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+// #define USE_I2C
+
+#define TAPPING_FORCE_HOLD
+#define TAPPING_TERM 100
+
+#ifdef RGBLIGHT_ENABLE
+    #undef RGBLED_NUM
+    #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/crkbd/keymaps/art4ride_pimoroni_trackball/config.h:38]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/crkbd/keymaps/art4ride_pimoroni_trackball/keymap.c [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+*/
+
+#include QMK_KEYBOARD_H
+#ifdef POINTING_DEVICE_ENABLE
+#include "pointing_device.h"
+#endif
+
+#ifdef PIMORONI_TRACKBALL_ENABLE
+#include "drivers/sensors/pimoroni_trackball.h"
+#endif
```
Comment (drashna) [inline keyboards/crkbd/keymaps/art4ride_pimoroni_trackball/keymap.c:26]:
```suggestion
#ifdef POINTING_DEVICE_ENABLE
#    include "pointing_device.h"
#    ifdef PIMORONI_TRACKBALL_ENABLE
#        include "drivers/sensors/pimoroni_trackball.h"
#    endif
#endif
```

File: keyboards/crkbd/keymaps/art4ride_pimoroni_trackball/keymap.c [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+void set_keylog(uint16_t keycode, keyrecord_t *record) {
+  char name = ' ';
+    if ((keycode >= QK_MOD_TAP && keycode <= QK_MOD_TAP_MAX) ||
+        (keycode >= QK_LAYER_TAP && keycode <= QK_LAYER_TAP_MAX)) { keycode = keycode & 0xFF; }
+  if (keycode < 60) {
+    name = code_to_name[keycode];
+  }
+
+  // update keylog
+  snprintf(keylog_str, sizeof(keylog_str), "%dx%d, k%2d : %c",
```
Comment (drashna) [inline keyboards/crkbd/keymaps/art4ride_pimoroni_trackball/keymap.c:133]: `snprintf` can take up 1.5kb of firmware space, and an unknown about of memory. 

You can do some fancy char manipulation to generate the numbers using a lot less firmware space, a la: 
https://github.com/qmk/qmk_firmware/blob/a2d43ca024d8e840c882cc72e80e9eb6b70384f1/users/drashna/oled_stuff.c#L302-L318

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**